ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -130,6 +130,58 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 		}
 		return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
+	case rayv1.JobDeploymentStatusInitializing:
+		var rayClusterInstance *rayv1.RayCluster
+		if rayClusterInstance, err = r.getOrCreateRayClusterInstance(ctx, rayJobInstance); err != nil {
+			return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
+		}
+		// If there is no cluster instance and no error, suspend the job deployment
```

Comment:
when would this case happen? Why wouldnt `getOrCreateRayClusterInstance` fail with err if no cluster instance was created

Reply:
Good question. Currently, the only scenario in which the function `getOrCreateRayClusterInstance` returns nil, nil is when `suspend` is set to true. It is not that straight-forward. I may refactor it in the following PRs.

https://github.com/ray-project/kuberay/blob/448e33d9b1787617a14d7d891b348c9c4b8cdf24/ray-operator/controllers/ray/rayjob_controller.go#L578-L581